oss_src/unity/lib/gl_sarray.cpp

```diff
@@ -28,9 +28,11 @@ static graphlab::mutex reader_shared_ptr_lock;
  */
 flex_type_enum infer_type_of_list(const std::vector<flexible_type>& vec) {
   std::set<flex_type_enum> types;
+  flex_type_enum last_type = flex_type_enum::UNDEFINED;
   for (const flexible_type& val: vec) {
-    if (val.get_type() != flex_type_enum::UNDEFINED) {
+    if(val.get_type() != last_type && val.get_type() != flex_type_enum::UNDEFINED) {
       types.insert(val.get_type());
+      last_type = val.get_type();
```

Comment:
last_type doesn't actually do very much here. This is an optimization and should be documented as such.